addons/account_payment/models/payment_transaction.py

```diff
@@ -196,15 +196,16 @@ def _log_message_on_linked_documents(self, message):
         :return: None
         """
         self.ensure_one()
-        self = self.with_user(SUPERUSER_ID)  # Log messages as 'OdooBot'
+        author = self.env.user.partner_id.id if self.env.uid == SUPERUSER_ID else self.partner_id.id
```

Comment:
I'm nitpicking but either you take the id and call the variable `author_id`, either you're taking the record and then you can call it `author` (and then do `author.id` in `message_post`. Not a mix :slightly_smiling_face: 
You took the record/author in the other file, please be consistent.